### PR TITLE
[WFLY-12686] Upgrade MicroProfile Metrics 2.2.1

### PR DESCRIPTION
**Pull request by jmesnil**
JIRA: https://issues.jboss.org/browse/WFLY-12686

### CODE REVIEW

File: microprofile/WFLY-12686_upgrade_microprofile_metrics_2.2.0.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+* Upgrade smallrye-metrics to 2.3.0
+* Run any MicroProfile Metrics test in WildFly integration test suite
+* Run the Eclipse MicroProfile Metrics TCK without failures
+* Add https://github.com/thorntail/thorntail/tree/master/testsuite/testsuite-microprofile-metrics[tests from Thorntail test suite that relates to MicroProfile Metrics] which is not already covered by the MicroProfile TCK or WildFly integration test suite
+
+== Implementation Plan
+
+== Test Plan
+
+* Add https://github.com/thorntail/thorntail/tree/master/testsuite/testsuite-microprofile-metrics[tests from Thorntail test suite that relates to MicroProfile Metrics] which is not already covered by the MicroProfile TCK or WildFly integration test suite. These new test cases will be added to WildFly basic ntegration test suite
```
Comment (jmesnil) [inline microprofile/WFLY-12686_upgrade_microprofile_metrics_2.2.0.adoc]: After reviewing these tests, I have not seen any behaviour that is not already covered by WildFly test suite or MicroProfile Metrics TCK.

Comment (istraka) [reply microprofile/WFLY-12686_upgrade_microprofile_metrics_2.2.0.adoc]: Thanks. Update the analysis then, please.

Comment (jmesnil) [reply microprofile/WFLY-12686_upgrade_microprofile_metrics_2.2.0.adoc]: done

**Comment by jmesnil** [comment]
@istraka I've updated the dev analysis.

**Comment by jmesnil** [comment]
@istraka I've updated the dev analysis https://github.com/wildfly/wildfly-proposals/pull/259/files#diff-21f8d6fe54a931d8213026e155da7e5aR50-R52

**Comment by istraka** [comment]
Thanks @jmesnil!

File: microprofile/WFLY-12686_upgrade_microprofile_metrics_2.2.0.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+== Non-Requirements
+
+* Using non-reusable metrics with same name across multiple deployments and/or subdeployments is not supported.
+* https://github.com/eclipse/microprofile-metrics/blob/master/spec/src/main/asciidoc/architecture.adoc#usage-of-microprofile-metrics-in-application-servers-with-multiple-applications[Recommendation for multiple applications] is not supported. Applications that wants to be supported in a multiple deployment state should provide a `mp.metrics.appName` configuration properties in the `META-INF/microprofile-config.properties` that is bundled in the archive.
+
+== Implementation Plan
+
+== Test Plan
+
+* Add https://github.com/thorntail/thorntail/tree/master/testsuite/testsuite-microprofile-metrics[tests from Thorntail test suite that relates to MicroProfile Metrics] which is not already covered by the MicroProfile TCK or WildFly integration test suite. These new test cases will be added to WildFly basic ntegration test suite
```
Comment (rhusar) [inline microprofile/WFLY-12686_upgrade_microprofile_metrics_2.2.0.adoc]: s/ntegration/integration

**Pull request merged**